Never store requests

pakrym-oai opened the pull request:
When item ids are sent to Responses API it will load them from the database ignoring the provided values. This adds extra latency. 

Not having the mode to store requests also allows us to simplify the code.

## Breaking change

The `disable_response_storage` configuration option is removed.

pakrym-oai committed:
Never store requests

aibrahim-oai commented on codex-rs/core/src/config.rs:
```diff
@@ -78,11 +78,7 @@ pub struct Config {
-    /// who have opted into Zero Data Retention (ZDR).
-    pub disable_response_storage: bool,
-
+    // Removed: server-side response storage control. Responses are never stored.
```
Do we need this comment?

aibrahim-oai commented on codex-rs/core/src/config.rs:
```diff
@@ -417,11 +413,7 @@ pub struct ConfigToml {
-    /// who have opted into Zero Data Retention (ZDR).
-    pub disable_response_storage: Option<bool>,
-
+    // Removed: disable_response_storage. Responses are never stored.
```
Do we need this comment?

aibrahim-oai commented on codex-rs/core/tests/suite/client.rs:
```diff
@@ -666,10 +659,7 @@ async fn prefers_apikey_when_config_prefers_apikey_even_with_chatgpt_tokens() {
-        request_body["store"].as_bool().unwrap(),
-        "store should be true for API key auth"
-    );
+    // store flag removed; responses are never stored.
```
same

pakrym-oai committed:
core: remove disable_response_storage flag and related logic

Remove the deprecated disable_response_storage configuration flag and all related logic, comments, and documentation references. The system now never stores responses, fully supporting OpenAI Zero Data Retention (ZDR) organizations by default. Update tests, config parsing, and request structures to reflect this change.

pakrym-oai committed:
config: remove obsolete comment regarding server-side response storage
